### PR TITLE
feat(skip-dialog): Enhance popup usability and design (#118)

### LINKED ISSUE #118: Enhance popup usability and design
> **Is your enhancement proposal related to a problem? Please describe.**
> The current popups ("Welcome to the new Phonebook!", "We need your help", "We've updated to a newer Version") can be very annoying if you don't safe cookies.
> 
> **Describe the solution you'd like**
> A popup like "snack-bar" which will open from one site and could be easy closed.
> 
> 
> > Issue #43 can be fixed here

**UPDATE:** The user will see "Welcome-Notification" and "Bug-Report-Dialog" on the first start of the page. If someone don't allow localstorage he will see this dialogs everytime. We should find a way to stop this.
--> There is no way.

## Idea to don't show the dialogs for users with no allowed cookies:
An workaround to don't show the dialogs every time could another url e.g. `phonebook.com/skip_dialogs` which then do not show the welcome-dialog and the major-update dialog. The bug-report-dialog should still be shown.

Probably could easy implement with setting the status before other lifecycle events with https://www.ngxs.io/concepts/actions or by just don't check the values in localstorage on the above url.

**Checklist**
- [x] Provide a new url-parameter (e.g. ?skip_dialog="true")
- [x] If this url is called the welcome popup will not be shown 
- [x] ..the "release-notification-dialog" on the second page load should also not be shown
- [x] Inform about the new url on the information-page and on the welcome-dialog

### PR DESCRIPTION
**Pull request by FrankLambrette**
Informations about the skip dialog-logic can now be found at the buttom of the welcome dialog. On Click on the information icon, users can directly follow the recommended url with click on the action. Snackbar will automaticly close after 6 seconds.

Should resolves #118 

I am probably not ready to change something until 07.10.

### CODE REVIEW

File: Phonebook.Frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -128,4 +149,5 @@ export class AppComponent implements OnInit {
       this.router.navigateByUrl('/');
     });
   }
+  public ngOnDestroy(): void {}
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: nope, that is required for `untilComponentDestroyed()`

File: Phonebook.Frontend/src/app/app.component.ts [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
 import { runtimeEnvironment } from 'src/environments/runtime-environment';
+import { untilComponentDestroyed } from 'ng2-rx-componentdestroyed';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy {
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: nope

File: Phonebook.Frontend/src/app/app.component.ts [hunk 3]
```diff
@@ -1,24 +1,28 @@
 import { Platform } from '@angular/cdk/platform';
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: nope

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html [hunk 4]
```diff
@@ -117,6 +117,12 @@ <h3 mat-dialog-title class="pb-color-primary" i18n="Display-notificationDialog|W
     <p i18n="Display-notificationDialog|Cookie notice@@WelcomeCookieNotice">
       This site uses cookies to ensure the basic functionality of the app.
     </p>
+    <div class="cookie-notices">
+      <p i18n="Display-notificationDialog|Annoyed by cookies notice@@PageInformationNewUrlNoCookiesText">
+        Annoyed by the start-dialogs? Klick here:
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]:
```suggestion
        Annoyed by the start-dialogues? Click here:
```

File: Phonebook.Frontend/src/app/app.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mschwrdtnr) [reply Phonebook.Frontend/src/app/app.component.ts]: Sure, but `untilComponentDestroyed()` is not used

File: Phonebook.Frontend/src/app/app.component.ts [hunk 5]
```diff
@@ -95,6 +100,17 @@ export class AppComponent implements OnInit {
     //   });
     // }
 
+    //queryParamMap did not work, because condition if dialog should open must wait for subscription
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: Than you should use the snapshot https://angular.io/api/router/ActivatedRouteSnapshot

File: Phonebook.Frontend/src/app/app.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: than lets see how the refactoring goes :)

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 6]
```diff
@@ -14,10 +18,48 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
...
   }
 
+  public openSnackBar() {
+    const url = this.sanitizer.bypassSecurityTrustHtml(
+      '<a href="' + window.location.href + '?skip_dialog=true">Link</a>'
+    );
+    const snackBarRef = this.snackBar
+      .open(
+        this.i18n({
+          meaning: 'Display-New-Url-Advice',
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: You dont need to set `-` Meaning is just text to explain more or give context to the translator (e.g. by referring the component name)

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html (hunk 4 above)
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: Maybe rephrase the whole sentence like "You deactivated cookies and keep getting those dialogues?"

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html [hunk 7]
```diff
@@ -117,6 +117,12 @@ <h3 mat-dialog-title class="pb-color-primary" i18n="Display-notificationDialog|W
...
+      </p>
+      <mat-icon class="announcement" (click)="openSnackBar()">announcement</mat-icon>
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: me too, there should be no click here: <button>. The text itself should already be the button.
Maybe rephrase to a description of what the button is doing. 
"Deactivate Startup Dialogs"

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.scss [hunk 8]
```diff
@@ -23,6 +23,19 @@ h3 {
   margin: 0 0 0 auto;
 }
 
+.announcement {
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.scss]: yep, use this https://material.angular.io/components/button/overview

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 9]
```diff
@@ -14,10 +18,48 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
...
+    private sanitizer: DomSanitizer
+  ) {
     store.dispatch(new SetDisplayedNotificationVersion(DisplayNotificationDialog.version));
   }
 
+  public openSnackBar() {
+    const url = this.sanitizer.bypassSecurityTrustHtml(
+      '<a href="' + window.location.href + '?skip_dialog=true">Link</a>'
+    );
+    const snackBarRef = this.snackBar
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: I would prefer that the user is navigated on the first click in the notification dialog. 
But I like the idea of the snackbar. Maybe doing it the other way around would be a great idea: 
1. User with disabled cookies clicks on the link in the dialog. (there should also be a note about saving this url as a favourite in order to keep the setting)
2. Gets redirected to the page with query Parameters for disabling dialogs
3. A snackBar is displayed Informing the user that he is having a degraded user experience by not getting any information about updates and can switch back with a click on the button

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 10]
```diff
@@ -14,10 +18,48 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
...
+          meaning: 'Go-to-New-Url-Advice',
+          description: 'Message for following no cookie url',
+          id: 'PageInformationNewUrlNoCookiesUrl',
+          value: 'Go to Url'
+        }),
+        { duration: 6000 }
+      )
+      .onAction()
+      .subscribe(() => {
+        this.router.navigateByUrl('/?skip_dialog=true');
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: because of the lines below, this line should be kept. Remove the two below

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 11]
```diff
@@ -14,10 +18,48 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
...
+        console.log(window.location);
+        window.location.href = window.location.href + '?skip_dialog=true';
```
[the remark this answers is not in the document]
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: Can you elaborate? People should be smart enough that they are saving the page they are viewing as a favourite.

[the remark this answers is not in the document]
Comment (mschwrdtnr) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: I also hope so 😄

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 12]
```diff
@@ -14,10 +18,48 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
...
+          value:
+            'Add skip_dialog=true as parameter to url for skipping start dialogs if cookies are deleted after each session.'
+        }),
+        this.i18n({
+          meaning: 'Go-to-New-Url-Advice',
+          description: 'Message for following no cookie url',
+          id: 'PageInformationNewUrlNoCookiesUrl',
+          value: 'Go to Url'
+        }),
+        { duration: 6000 }
```
[the remark this answers is not in the document]
Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: I think for the new behaviour (open snackbar after no-cookie-url is loaded) the duration should be fine, what do you think?

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: `this.router.navigateByUrl('/?skip_dialog=true');` -->will set the router url, but page does not reload, so the user won't realy notice a change...i think this should be removed (this is useless at the moment, because site will reload in next line) 

`window.location.href = window.location.href + '?skip_dialog=true';` -->will reload the page with the parameter --> user notices the change, i think we should keep this

**Comment by DanielHabenicht** [comment]
That's way better, without the extra dialog. 👍 

If you think that reloading the site is more intuitive to the user - I doubt it. 
How about making it really clear by providing a snack bar with "Save the site URL as a favourite now in order to not get any more dialogs"? 
In the end, there should be no difference between navigating in the app, for various reasons (clean code -> angular has the router for navigation; arising edge cases; inconsistent app behaviour; more network traffic)

**Comment by mschwrdtnr** [comment]
@all-contributors please add @FrankLambrette for code

**Comment by FrankLambrette** [comment]
> That's way better, without the extra dialog. 👍
> 
> If you think that reloading the site is more intuitive to the user - I doubt it.
> How about making it really clear by providing a snack bar with "Save the site URL as a favourite now in order to not get any more dialogs"?
> In the end, there should be no difference between navigating in the app, for various reasons (clean code -> angular has the router for navigation; arising edge cases; inconsistent app behaviour; more network traffic)

Changed the workflow as mentioned in the last commit.

File: Phonebook.Frontend/src/app/app.component.ts [hunk 13]
```diff
@@ -95,8 +99,19 @@ export class AppComponent implements OnInit {
...
 
+    //if skip_cookies is set, dont show dialogs
+    if (this.skipDialog) {
+      this.store.dispatch(new SetDisplayedNotificationVersion(DisplayNotificationDialog.version));
+    }
+    //subscribe on query param changes, if changed open snackbar
+    this.activatedRoute.queryParamMap.pipe(untilComponentDestroyed(this)).subscribe(queryParamMap => {
+      if (queryParamMap.get('skip_dialog') === 'true') {
+        this.skipDialog = true;
+        this.openSnackBar();
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/app.component.ts]: I would just rename the function to something more descriptive. `openUrlSnackbar` or something like

File: Phonebook.Frontend/src/app/app.component.ts [hunk 14]
```diff
@@ -128,4 +143,29 @@ export class AppComponent implements OnInit {
       this.router.navigateByUrl('/');
     });
   }
+  public openSnackBar() {
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/app.component.ts]: I would just rename the function to something more descriptive. `openUrlSnackbar` or something like

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts [hunk 15]
```diff
@@ -14,10 +15,13 @@ import { SetDisplayedNotificationVersion } from 'src/app/shared/states';
 })
 export class DisplayNotificationDialog implements OnInit, OnDestroy {
   public static version: number = 1;
-  constructor(public dialogRef: MatDialogRef<DisplayNotificationDialog>, store: Store) {
+  constructor(public dialogRef: MatDialogRef<DisplayNotificationDialog>, store: Store, private router: Router) {
     store.dispatch(new SetDisplayedNotificationVersion(DisplayNotificationDialog.version));
   }
-
+  public skipDialoges() {
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.ts]: rename to `skipDialogs()`

File: Phonebook.Frontend/src/app/app.component.ts [hunk 16]
```diff
@@ -128,4 +143,29 @@ export class AppComponent implements OnInit {
...
+        }),
+        { duration: 15000 }
+      )
+      .onAction()
+      .subscribe(() => {
+        //remove url parameter and close snackbar
+        this.router.navigateByUrl('/');
+      });
+  }
+  public ngOnDestroy(): void {}
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/app.component.ts]:
```suggestion
  public ngOnDestroy() {}
```

File: Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html [hunk 17]
```diff
@@ -112,11 +112,22 @@ <h3 mat-dialog-title class="pb-color-primary" i18n="Display-notificationDialog|W
...
       This site uses cookies to ensure the basic functionality of the app.
-    </p>
+    </span>
+    <span i18n="Display-notificationDialog|Annoyed by cookies notice@@PageInformationNewUrlNoCookiesText">
+      You deactivated cookies and keep getting those dialogs?
+    </span>
+    <button
+      i18n="Display-notificationDialog|Deactivate Startup Dialogues@@PageInformationNewUrlNoCookiesButton"
+      mat-button
+      (click)="skipDialoges()"
```
Comment (mschwrdtnr) [inline Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: rename to `skipDialogs()`

Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: Maybe we should find a general naming convention for that bacause @DanielHabenicht suggested "dialogues" in a previous comment. Of course you are right, "dialoges" is definitely not correct.

Comment (mschwrdtnr) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: than why not just `skipStartDialogs()` ? I think this is the best solution with best understanding

Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/shared/dialogs/display-notification-dialog/display-notification.dialog.html]: Because the officially spelling is "dialogues". But of course we can agree on changing all occurrences to "dialogs".

**Comment by DanielHabenicht** [comment]
@paule96 Why is the preview working here but not in the other PR? :D

File: Phonebook.Frontend/src/app/app.component.ts [hunk 18]
```diff
@@ -95,8 +99,19 @@ export class AppComponent implements OnInit {
     //   });
     // }
 
+    //if skip_cookies is set, dont show dialogs
+    if (this.skipDialog) {
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]: Can we add this directly to the if statements, which are opening the dialogs, below?

Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: What I mean is don't set the DisplayedNotificationVersion as is never displayed. Just don't call the functions opening the dialogs.

File: Phonebook.Frontend/src/app/app.component.ts [hunk 19]
```diff
@@ -95,8 +99,19 @@ export class AppComponent implements OnInit {
...
+      this.store.dispatch(new SetDisplayedNotificationVersion(DisplayNotificationDialog.version));
+    }
+    //subscribe on query param changes, if changed open snackbar
+    this.activatedRoute.queryParamMap.pipe(untilComponentDestroyed(this)).subscribe(queryParamMap => {
+      if (queryParamMap.get('skip_dialog') === 'true') {
+        this.skipDialog = true;
+        this.openSkippedDialogsSnackBar();
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]: Can we have two different Notifications? If you added the URL as favourite you keep gettings the call to action. 

I would propose to have two different messages: 
1. Speicher die Seiten-URL jetzt als Favorit um Startdialoge auch zukünftig zu überspringen. Bitte beachte: Mit dem gesetzten Url-Parameter werden keine Dialoge zu Updates oder Neuen Features angezeigt.
2. Dialoge sind deaktiviert. Bitte beachte das du keine Informationen zu Updates oder Neuen Features angezeigt bekommst.

File: Phonebook.Frontend/src/app/app.component.ts [hunk 20]
```diff
@@ -128,4 +143,29 @@ export class AppComponent implements OnInit {
...
+          description: 'Message for new no cookie url',
+          id: 'PageInformationNewUrlNoCookies',
+          value:
+            'Save the site URL as a favourite now in order to not get any more dialogs. Please notice: You wont get any information about updates or releases with the set url prameter.'
+        }),
+        this.i18n({
+          meaning: 'Restore Url',
+          description: 'Message for following no cookie url',
+          id: 'PageInformationNewUrlNoCookiesUrl',
+          value: 'Reset Url'
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]: Können wir uns hier einen anderen Text ausdenken, z.B. "Dialoge wieder anzeigen" oder einfach "Rückgängig"? 
Ich denke mal der Standard Nutzer wird unter URL zurücksetzen nicht viel verstehen.

File: Phonebook.Frontend/src/app/app.component.ts (hunk 18 above)
Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/app.component.ts]: I could do so, but the url parameter gets lost on state change, so on reload the dialogs will appear again if DisplayedNotificationVersion is not set.

Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: Can't quite follow. Normal usage of the app never reloads the page. The user has to use the saved favourites link in order to not get the dialog... But I get what you are saying.

**Comment by FrankLambrette** [comment]
I changed the workflow as mentioned in the previous comments:
1 start dialog for open the page from favorite with url paremeter
2 start dialog after click on "deactivate startup dialogs
3 changed reset url to reset

Do you think it would be a good soloution to reopen a start dialog after cklick on "reset"? So the user who accidentally clicked on "deactivate startup dialogs" can view the start diaog again?

**Comment by FrankLambrette** [comment]
@DanielHabenicht please review this PR, thanks:)

File: Phonebook.Frontend/src/i18n/messages.de.xlf [hunk 21]
```diff
@@ -1238,6 +1254,19 @@ Diese Seite nutzt Cookies, um die Bedienung der Seite zu verbessern. Cookies wer
         <target state="translated">Gebäude</target>
         <note priority="1" from="description">Title of Table Column "Building"</note>
         <note priority="1" from="meaning">TableComponent</note>
+      </trans-unit><trans-unit id="PageInformationNewUrlNoCookies" datatype="html">
+        <source>Save the site URL as a favourite now in order to not get any more dialogs. Please notice: You wont get any information about updates or releases with the set url prameter.</source>
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/i18n/messages.de.xlf]:
```suggestion
        <source>Save the site URL as a favourite now in order to not get any more startup dialogs. Please notice: You wont get any information about updates or releases with the set url prameter.</source>
```

File: Phonebook.Frontend/src/app/app.component.ts [hunk 22]
```diff
@@ -128,4 +146,52 @@ export class AppComponent implements OnInit {
...
+      });
+  }
+  public openSkippedDialogsSnackBar() {
+    this.snackBar
+      .open(
+        this.i18n({
+          meaning: 'Warning no dialogs are shown',
+          description: 'Message to inform user that no dialogs will be shown',
+          id: 'PageInformationNoDialogs',
+          value: 'Dialogs are deactivated. Please notice: You wont get any information about updates or releases.'
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]:
```suggestion
          value: 'Startup-Dialogs are deactivated. Please notice: You wont get any information about updates or releases.'
```

File: Phonebook.Frontend/src/app/app.component.ts [hunk 23]
```diff
@@ -128,4 +146,52 @@ export class AppComponent implements OnInit {
...
   }
+  public openJustSkippedDialogsSnackBar() {
+    this.snackBar
+      .open(
+        this.i18n({
+          meaning: 'Display advice to new url',
+          description: 'Message for just set no cookie url',
+          id: 'PageInformationNewUrlNoCookies',
+          value:
+            'Save the site URL as a favourite now in order to not get any more dialogs. Please notice: You wont get any information about updates or releases with the set url prameter.'
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]:
```suggestion
            'Save the site URL as a favourite now in order to not get any more startup-dialogs. Please notice: You won't get any information about updates or releases with the set url parameter.'
```

Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: Please rerun `extract-i18n` afterwards.

**Comment by DanielHabenicht** [comment]
Last two changes! Then we can merge :)

File: Phonebook.Frontend/src/app/app.component.ts (hunk 23 above)
Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/app.component.ts]: Done!

File: Phonebook.Frontend/src/app/app.component.ts [hunk 24]
```diff
@@ -154,7 +154,7 @@ export class AppComponent implements OnInit, OnDestroy {
           description: 'Message for just set no cookie url',
           id: 'PageInformationNewUrlNoCookies',
           value:
-            'Save the site URL as a favourite now in order to not get any more dialogs. Please notice: You wont get any information about updates or releases with the set url prameter.'
+            'Save the site URL as a favourite now in order to not get any more startup-dialogs. Please notice: You won\'t get any information about updates or releases with the set url parameter.'
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]: You could also use these quotes `` ` `` so you don't have to write any special characters in your text: `` `Save the site URL as a favourite now in order to not get any more startup-dialogs. Please notice: You won't get any information about updates or releases with the set url parameter.` ``

Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: just for info (no need to commit it, if you don't want to

Comment (FrankLambrette) [reply Phonebook.Frontend/src/app/app.component.ts]: Thanks for your hint, i didn't know that:)
